tests/test_json.ml

```diff
@@ -0,0 +1,60 @@
+let read_file fname =
+  (* the tests are run from the _build/default/tests directory *)
+  Fpath.(v ".." / ".." / ".." / "tests" / fname)
+  |> Fpath.to_string |> Yojson.Basic.from_file
+
+let dune_release_testing_json = read_file "dune-release-testing.json"
+
+let upload_response_json = read_file "upload-response.json"
+
+let create_release_response_json = read_file "create-release-response.json"
+
+let pull_request_response_json = read_file "pull-request-response.json"
+
+let archive_upload_url () =
+  let check json expected =
+    match Dune_release.Json.Upload_response.browser_download_url json with
+    | Ok actual -> Alcotest.(check string) __LOC__ actual expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check dune_release_testing_json
+    "https://github.com/NathanReb/dune-release-testing/releases/download/v0.0.0/dummy-v0.0.0.tbz";
+  check upload_response_json
+    "https://github.com/octocat/Hello-World/releases/download/v1.0.0/example.zip"
+
+let release_id () =
+  let check json expected =
+    match Dune_release.Json.Release_response.release_id json with
+    | Ok actual -> Alcotest.(check int) __LOC__ actual expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check create_release_response_json 1
+
+let upload_url () =
+  let check json expected =
+    match Dune_release.Json.Release_response.upload_url json with
+    | Ok actual -> Alcotest.(check string) __LOC__ actual expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check create_release_response_json
+    "https://uploads.github.com/repos/octocat/Hello-World/releases/1/assets{?name,label}"
+
+let html_url () =
+  let check json expected =
+    match Dune_release.Json.Pull_request_response.html_url json with
+    | Ok (`Url actual) -> Alcotest.(check string) __LOC__ actual expected
+    | Ok `Already_exists ->
+        Alcotest.(check string) __LOC__ "ALREADY_EXISTS" expected
+    | Error (`Msg msg) -> Alcotest.fail msg
+  in
+  check pull_request_response_json
+    "https://github.com/octocat/Hello-World/pull/1347"
+
+let suite =
+  ( "Json",
```

Comment:
This is all specific to github's API so how about moving them to a `Github.V3_api` submodule or `Github_v3_api` module rather than `Json`?

I think it can be slightly misleading as I naturally hope to find generic json helpers there.

Another advantage of introducing a github v3-api specific module is that we can use it to abstract requests and responses in the future to improve the code handling the API interactions.

What do you think?

Reply:
I agree, I renamed the module into `Github_v3_api`